airbyte-webapp/src/components/ui/Icon/types.ts

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+
+import { ReactComponent as ArrowRightIcon } from "./icons/arrowRightIcon.svg";
+import { ReactComponent as CreditsIcon } from "./icons/creditsIcon.svg";
+import { ReactComponent as CrossIcon } from "./icons/crossIcon.svg";
+import { ReactComponent as DocsIcon } from "./icons/docsIcon.svg";
+import { ReactComponent as GAIcon } from "./icons/gAIcon.svg";
+import { ReactComponent as InfoIcon } from "./icons/infoIcon.svg";
+import { ReactComponent as MinusIcon } from "./icons/minusIcon.svg";
+import { ReactComponent as ModificationIcon } from "./icons/modificationIcon.svg";
+import { ReactComponent as MoonIcon } from "./icons/moonIcon.svg";
+import { ReactComponent as PauseIcon } from "./icons/pauseIcon.svg";
+import { ReactComponent as PencilIcon } from "./icons/pencilIcon.svg";
+import { ReactComponent as PlayIcon } from "./icons/playIcon.svg";
+import { ReactComponent as PlusIcon } from "./icons/plusIcon.svg";
+import { ReactComponent as RotateIcon } from "./icons/rotateIcon.svg";
+
+export type IconType =
+  | "arrowRight"
+  | "credits"
+  | "cross"
+  | "docs"
+  | "ga"
+  | "info"
+  | "minus"
+  | "modification"
+  | "moon"
+  | "pause"
+  | "pencil"
+  | "play"
+  | "plus"
+  | "rotate";
+
+export interface IconProps {
+  type: IconType;
+  className?: string;
+  height?: number | string;
+  width?: number | string;
+  // We need to consider the implementation of a mechanism with predefined colors and sizes
```

Comment:
I think we should at least build in the `size` property for now, instead of adding a `height`/`width`, that we'd then later once that icon is already used need to change in all consuming places. I think the color could potentially be part of a separate PR.

Reply:
@timroes, should it be something similar to other components, for instance, property `large` (height 25px/width 25px), `medium` (height 15px/width 15px)...?

Reply:
Yes, I'd suggest we use the size abbreviations we use in other places as well (`sm`, `md`, `lg`, `xl`, ...). I don't have a clear idea what they might match to. @Upmitt @edmundito do you know if we have standardized icon sizes in our designs usually that we can match them to?

Reply:
We have not defined the icon sizes yet, will follow up about this but this should not block the PR.